Code/Framework/AzCore/AzCore/Serialization/SerializeContext.cpp

```diff
@@ -3178,17 +3178,17 @@ namespace AZ
 
     void SerializeContext::PerModuleGenericClassInfo::Cleanup()
     {
-        decltype(m_moduleLocalGenericClassInfos) genericClassInfoContainer = AZStd::move(m_moduleLocalGenericClassInfos);
-        decltype(m_serializeContextSet) serializeContextSet = AZStd::move(m_serializeContextSet);
+        auto genericClassInfoContainer = AZStd::move(m_moduleLocalGenericClassInfos);
+        auto serializeContextSet = AZStd::move(m_serializeContextSet);
         // Un-reflect GenericClassInfo from each serialize context registered with the module
         // The SerailizeContext uses the SystemAllocator so it is required to be ready in order to remove the data
         if (AZ::AllocatorInstance<AZ::SystemAllocator>::IsReady())
         {
             for (AZ::SerializeContext* serializeContext : serializeContextSet)
             {
-                for (const AZStd::pair<AZ::Uuid, AZ::GenericClassInfo*>& moduleGenericClassInfoPair : genericClassInfoContainer)
+                for (const auto& [specializedTypeId, genericClassInfo] : genericClassInfoContainer)
```

Comment:
It is inconsistent to change this iterator over the genericClassInfoContainer, but not the one right below this if block.

The for loop below should use a structured binding to be consistent with this change.

Reply:
Fixed.